build: set up aarch64 cross-compilation toolchain

nvb opened the pull request:
This change configures the cockroachdb/builder docker image to be able
to target `aarch64-unknown-linux-gnueabi`. It does this by adding a new
crosstool-ng toolchain. This toolchain is almost identical to
`x86_64-unknown-linux-gnu` except that it targets the aarch64
architecture. It also uses the Linux 3.7.10 kernel instead of Linux
2.6.32 because Linux 3.7 was the first to support 64-bit ARM.

Release note (build change): The Docker builder image is now able to
cross-compile release aarch64 binaries.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/24354)
<!-- Reviewable:end -->

nvb committed:
build: set up aarch64 cross-compilation toolchain

This change configures the cockroachdb/builder docker image to be able
to target `aarch64-unknown-linux-gnueabi`. It does this by adding a new
crosstool-ng toolchain. This toolchain is almost identical to
`x86_64-unknown-linux-gnu` except that it targets the aarch64
architecture. It also uses the Linux 3.7.10 kernel instead of Linux
2.6.32 because Linux 3.7 was the first to support 64-bit ARM.

Release note (build change): The Docker builder image is now able to
cross-compile release aarch64 binaries.

benesch commented on build/builder.sh:
```diff
@@ -3,7 +3,9 @@
-version=20180220-200046
+#version=20180220-200046
+# WIP update before merging.
+version=latest
```
don't forget about this!

nvanbenschoten replied:
Done.

benesch reviewed (approved):
LGTM modulo updating the builder!

benesch commented:
If CI is happy so am I!

nvanbenschoten commented:
TFTR!
